opennms-webapp/src/main/webapp/heatmap/heatmap-box.jsp

```diff
@@ -118,80 +118,84 @@
+                    <%
+                      }
+                    %>
+                };
 
-                if (height < 0) {
-                    height = $(window).width();
-                }
+                var url = "<%=url%>";
+                var children;
 
-                height = Math.min(height, $(window).width());
+                function refresh() {
+                    var height = $(window).height() - 105 - $("#treemap").offset().top;
 
-                $("#treemap").treemap({
-                    "dimensions": [
-                        $("#treemap").width(),
-                        height
-                    ],
-                    "colorStops": [
-                        {"val": 1.0, "color": "#CC0000"},
-                        {"val": 0.4, "color": "#FF3300"},
-                        {"val": 0.2, "color": "#FF9900"},
-                        {"val": 0.1, "color": "#FFCC00"},
-                        {"val": 0.0, "color": "#336600"}
-                    ],
-                    "labelsEnabled": true,
-                    "nodeData": {
-                        "id": "<%=heatmap%>",
-                        "children": children
+                    if (height < 0) {
+                        height = $(window).width();
                     }
-                }).bind('treemapclick', mouseclickHandler);
-            }
 
-            $(window).resize(function() {
-                refresh();
-            });
+                    height = Math.min(height, $(window).width());
 
-            $(document).ready(function () {
-                $.getJSON(url, function (data) {
-                    children = data.children;
+                    $("#treemap").treemap({
+                        "dimensions": [
+                            $("#treemap").width(),
+                            height
+                        ],
+                        "colorStops": [
+                            {"val": 1.0, "color": "#CC0000"},
+                            {"val": 0.4, "color": "#FF3300"},
+                            {"val": 0.2, "color": "#FF9900"},
+                            {"val": 0.1, "color": "#FFCC00"},
+                            {"val": 0.0, "color": "#336600"}
+                        ],
+                        "labelsEnabled": true,
+                        "nodeData": {
+                            "id": "<%=heatmap%>",
+                            "children": children
+                        }
+                    }).bind('treemapclick', mouseclickHandler);
+                }
+
+                $(window).resize(function() {
                     refresh();
                 });
+
+                $(document).ready(function () {
+                    $.getJSON(url, function (data) {
+                        children = data.children;
+                        refresh();
+                    });
+                });
             });
-        });
+        }
+
+        setTimeout(initialize, 250);
```

Comment:
I found this workaround on stackoverflow.com for these "$.widget is not a function" errors. But it also works by encapsulating the code inside a window.onload function. So, I changed this...